src/Graphics/SamplerStateCollection.cs

```diff
@@ -38,20 +35,21 @@ public sealed class SamplerStateCollection
 		#region Private Variables
 
 		private readonly SamplerState[] samplers;
-		private readonly Queue<int> modifiedSamplers;
+		private readonly bool[] modifiedSamplers;
 
 		#endregion
 
 		#region Internal Constructor
 
 		internal SamplerStateCollection(
 			int slots,
-			Queue<int> modSamplers
+			bool[] modSamplers
 		) {
 			samplers = new SamplerState[slots];
 			modifiedSamplers = modSamplers;
 			for (int i = 0; i < samplers.Length; i += 1)
 			{
+				// FIXME: Shouldn't we be setting modifiedSamplers here?
```

Comment:
Only if the GL default sampler parameters doesn’t match. Thankfully this appears to be consistent for every target, and of course the default init could be defined at IGLDevice initialization.